python/cugraph/cugraph/gnn/data_loading/bulk_sampler.py

```diff
@@ -0,0 +1,156 @@
+
+    def __init__(
+        self,
+        output_path: str,
+        graph,
+        saturation_level: int = 200_000,
+        rank: int = 0,
+        **kwargs,
+    ):
+        """
+        Constructs a new BulkSampler
+
+        Parameters
+        ----------
+        output_path: str
+            The directory where results will be stored.
+        graph: cugraph.Graph
+            The cugraph graph to operate upon.
+        saturation_level: int (optional, default=200,000)
+            The number of samples that can be made within a single call.
+        rank: int (optional, default=0)
+            The rank of this sampler.  Used to isolate this sampler from
+            others that may be running on other nodes.
+        kwargs: kwargs
+            Keyword arguments to be passed to the sampler (i.e. fanout).
+        """
+        self.__output_path = output_path
+        self.__graph = graph
+        self.__saturation_level = saturation_level
+        self.__rank = rank
+        self.__batches = None
+        self.__sample_call_args = kwargs
+
+    @property
+    def rank(self) -> int:
+        return self.__rank
+
+    @property
+    def saturation_level(self) -> int:
+        return self.__saturation_level
+
+    @property
+    def size(self) -> int:
+        if self.__batches is None:
+            return 0
+        else:
+            return len(self.__batches)
+
+    def add_batches(
+        self,
+        df: Union[cudf.DataFrame, dask_cudf.DataFrame],
+        start_col_name: str,
+        batch_col_name: str,
+    ) -> None:
+        """
+        Adds batches to this BulkSampler.
+
+        Parameters
+        ----------
+        df: cudf.DataFrame or dask_cudf.DataFrame
+            Contains columns for vertex ids, batch id
+        start_col_name: str
+            Name of the column containing the start vertices
+        batch_col_name: str
+            Name of the column containing the batch ids
+
+        Returns
+        -------
+        """
+        df = df.rename(
+            columns={
+                start_col_name: self.start_col_name,
+                batch_col_name: self.batch_col_name,
+            }
+        )
+
+        if self.__batches is None:
+            self.__batches = df
+        else:
+            if isinstance(df, type(self.__batches)):
+                self.__batches = self.__batches.append(df)
+            else:
+                raise TypeError(
+                    "Provided batches must match the dataframe"
+                    " type of previous batches!"
+                )
+
+        if self.size >= self.saturation_level:
+            self.flush()
+
+    def flush(self) -> None:
+        """
+        Computes all uncomputed batches
+        """
+        end = min(self.__saturation_level, len(self.__batches))
+
+        sample_fn = (
+            cugraph.dask.uniform_neighbor_sample
+            if isinstance(self.__batches, dask_cudf.DataFrame)
+            else cugraph.uniform_neighbor_sample
```

Comment:
We should do this based on what the Graph object is rather than the sampling list . 
```suggestion
            cugraph.dask.uniform_neighbor_sample
            if isinstance(self.__batches, dask_cudf.DataFrame)
            else cugraph.uniform_neighbor_sample
```
```suggestion
            cugraph.uniform_neighbor_sample
            if isinstance(self.__graph._plc_graph, pylibcugraph.graphs.SGGraph)
            else cugraph.dask.uniform_neighbor_sample
```